test/performance/odp_ipsecfwd.c

```diff
@@ -0,0 +1,1307 @@
+
+			printf(")\n");
+		}
+	}
+}
+
+static void print_usage(void)
+{
+	odp_pool_capability_t pool_capa;
+	odp_ipsec_capability_t ipsec_capa;
+
+	if (odp_pool_capability(&pool_capa) < 0) {
+		ODPH_ERR("Error querying pool capabilities\n");
+		return;
+	}
+
+	if (odp_ipsec_capability(&ipsec_capa) < 0) {
+		ODPH_ERR("Error querying IPsec capabilities\n");
+		return;
+	}
+
+	printf("\n"
+	       "Simple IPsec performance tester. Forward and process plain and ipsec packets.\n"
+	       "\n"
+	       "Examples:\n"
+	       "    %s -i ens9f1 -s /etc/odp/sa.conf -f /etc/odp/fwd.conf\n"
+	       "\n"
+	       "    With sa.conf containing, for example:\n"
+	       "        0 222 192.168.1.10 192.168.1.16 4 jWnZr4t7w!zwC*F- 0 2"
+	       " n2r5u7x!A%%D*G-KaPdSg 0 12\n"
+	       "\n"
+	       "    With fwd.conf containing, for example:\n"
+	       "        192.168.1.0 24 ens9f1 aa:bb:cc:dd:11:22\n"
+	       "\n"
+	       "Usage: %s [options]\n"
+	       "\n"
+	       "  -i, --interfaces    Ethernet interfaces for packet I/O, comma-separated,\n"
+	       "                      no spaces.\n"
+	       "  -n, --num_pkts      Number of packet buffers allocated for packet I/O pool.\n"
+	       "                      %u by default.\n"
+	       "  -l, --pkt_len       Maximum size of packet buffers in packet I/O pool. %u by\n"
+	       "                      default.\n"
+	       "  -c, --count         CPU count, %u by default.\n"
+	       "  -m, --mode          Packet output mode.\n"
+	       "                          0: queued output (default)\n"
+	       "                          1: direct output\n"
+	       "  -s, --sa            SA configuration file. Individual SA configuration is\n"
+	       "                      expected to be within a single line, values whitespace\n"
+	       "                      separated:\n"
+	       "\n"
+	       "                          <line in file> Dir SPI SrcIPv4 DstIPv4 CipherAlgoIdx"
+	       " CipherKey CipherKeyExtra AuthAlgIdx AuthKey AuthKeyExtra ICVLen\n"
+	       "\n"
+	       "                      With combined algorithms, authentication data is ignored.\n"
+	       "                      Traffic is mapped to SAs based on UDP port: the port is\n"
+	       "                      used as the SPI. Non-zero Dir value declares an outbound\n"
+	       "                      SA whereas zero Dir value declares an inbound SA.\n"
+	       "\n"
+	       "                      Supported cipher and authentication algorithms:\n",
+	       PROG_NAME, PROG_NAME, MIN(pool_capa.pkt.max_num, PKT_CNT),
+	       MIN(pool_capa.pkt.max_len, PKT_SIZE), WORKER_CNT);
+	print_supported_algos(&ipsec_capa);
+	printf("  -f, --fwd_table     Forwarding configuration file. Individual forwarding\n"
+	       "                      configuration is expected to be within a single line,\n"
+	       "                      values whitespace separated:\n"
+	       "\n"
+	       "                          <line in file> DstIPv4 SubNetMask NetIf DstMac\n"
```

Comment:
I think it would be nicer to have <subnet>/<masklen> kind of syntax. I guess just adding the slash to the fscanf() string would do it.

Reply:
v4.

Reply:
Now with the syntax changed, this text should be updated. I also suggest using MaskLen instead of just Mask for the MaskLen Part. How about: IPv4Prefix/MaskLen?